### PR TITLE
move things out of mir/mod.rs

### PR DESCRIPTION
**Pull request by RalfJung**
This moves a bunch of things out of `mir/mod.rs`:
- all const-related stuff to a new file consts.rs
- all statement/place/operand-related stuff to a new file statement.rs
- all pretty-printing related stuff to pretty.rs

`mod.rs` started out with 3100 lines and ends up with 1600. :)

Also there was some pretty-printing stuff in terminator.rs, that also got moved to pretty.rs, and I reordered things in pretty.rs so that it can be grouped by functionality.

Only the commit "use pretty_print_const_value from MIR constant 'extra' printing" has any behavior changes; it resolves the issue of having a fancy and a very crude pretty-printer for `ConstValue`.

r? @oli-obk

### CODE REVIEW

**Comment by RalfJung** [comment]
I added two more commits that move things around a bit and clarify a commit; they can be reviewed separately (and they would conflict terribly if I made them a separate PR).

This bitrots quickly so I'd appreciate quick feedback. :)

File: compiler/rustc_codegen_cranelift/src/constant.rs [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 use rustc_data_structures::fx::{FxHashMap, FxHashSet};
 use rustc_middle::middle::codegen_fn_attrs::CodegenFnAttrFlags;
-use rustc_middle::mir::interpret::{
-    read_target_uint, AllocId, ConstValue, ErrorHandled, GlobalAlloc, Scalar,
+use rustc_middle::mir::{
+    interpret::{read_target_uint, AllocId, ErrorHandled, GlobalAlloc, Scalar},
+    ConstValue,
```
Comment (bjorn3) [inline compiler/rustc_codegen_cranelift/src/constant.rs:7]: Please split this into two separate imports.

**Review by oli-obk — APPROVED** [review]

**Comment by RalfJung** [comment]
@bors r=oli-obk p=1
(high bitrot risk)

**Comment by RalfJung** [comment]
Uh, it did not land but we got a perf result??

**Pull request merged**